src/internal/Subscription.ts

```diff
@@ -186,26 +136,18 @@ export class Subscription implements SubscriptionLike {
    *
    * All finalizer instances are removed to free up memory upon unsubscription.
    *
+   * TIP: In instances you're adding and removing _Subscriptions from other Subscriptions_, you should
+   * be sure to unsubscribe or otherwise get rid of the child subscription reference as soon as you remove it.
+   * The child subscription has a reference to the parent it was added to via closure. In most cases, this
+   * a non-issue, as child subscriptions are rarely long-lived.
+   *
    * @param teardown The finalizer to remove from this subscription
    */
   remove(teardown: Exclude<TeardownLogic, void>): void {
     this._finalizers?.delete(teardown);
-
-    if (teardown instanceof Subscription) {
-      teardown._removeParent(this);
-    }
   }
 }
 
-export const EMPTY_SUBSCRIPTION = Subscription.EMPTY;
-
-export function isSubscription(value: any): value is Subscription {
```

Comment:
This was only used in the Observable `subscribe` call, and honestly was more robust than was necessary.